GoogleSignIn/Tests/Unit/GIDAuthorizationResponseHelperTest.m

```diff
@@ -0,0 +1,169 @@
+#import "GoogleSignIn/Sources/GIDAuthorizationResponse/Implementations/GIDAuthorizationResponseHandler.h"
+
+#import "GoogleSignIn/Sources/Public/GoogleSignIn/GIDConfiguration.h"
+#import "GoogleSignIn/Sources/Public/GoogleSignIn/GIDSignIn.h"
+#import "GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h"
+#import "GoogleSignIn/Sources/GIDAuthFlow.h"
+
+#import "GoogleSignIn/Tests/Unit/OIDAuthorizationResponse+Testing.h"
+#import "GoogleSignIn/Tests/Unit/OIDAuthState+Testing.h"
+
+// The EMM support version
+static NSString *const kEMMVersion = @"1";
+
+@interface GIDAuthorizationResponseHelperTest : XCTestCase
+@end
+
+@implementation GIDAuthorizationResponseHelperTest
+
+- (void)testInitWithAuthorizationResponseHandler {
+  GIDAuthorizationResponseHandler *responseHandler =
+      [[GIDAuthorizationResponseHandler alloc] initWithAuthorizationResponse:nil
+                                                                  emmSupport:nil
+                                                                    flowName:GIDFlowNameVerify
+                                                               configuration:nil
+                                                                       error:nil];
+  GIDAuthorizationResponseHelper *responseHelper = 
+      [[GIDAuthorizationResponseHelper alloc] initWithAuthorizationResponseHandler:responseHandler];
+
+  XCTAssertNotNil(responseHelper);
+  XCTAssertNotNil(responseHelper.responseHandler);
+}
+
+- (void)testFetchTokenWithAuthFlow {
+  OIDAuthState *authState = [OIDAuthState testInstance];
+  GIDAuthorizationResponseHandlingFake *fakeHandler = 
+      [[GIDAuthorizationResponseHandlingFake alloc] initWithAuthState:authState error:nil];
+  GIDAuthorizationResponseHelper *responseHelper =
+      [[GIDAuthorizationResponseHelper alloc] initWithAuthorizationResponseHandler:fakeHandler];
+  GIDAuthFlow *authFlow = [[GIDAuthFlow alloc] init];
+
+  [responseHelper fetchTokenWithAuthFlow:authFlow];
+  XCTAssertNotNil(authFlow);
+  XCTAssertNotNil(authFlow.authState);
+  XCTAssertNil(authFlow.error);
+}
+
+- (void)testSuccessfulGenerateAuthFlowFromAuthorizationResponse {
+  OIDAuthorizationResponse *authorizationResponse = [OIDAuthorizationResponse testInstance];
+  GIDAuthorizationResponseHandler *responseHandler =
+      [[GIDAuthorizationResponseHandler alloc] initWithAuthorizationResponse:authorizationResponse
+                                                                  emmSupport:nil
+                                                                    flowName:GIDFlowNameVerify
+                                                               configuration:nil
+                                                                       error:nil];
+
+  GIDAuthFlow *authFlow = [responseHandler generateAuthFlowFromAuthorizationResponse];
+  XCTAssertNotNil(authFlow);
+  XCTAssertNotNil(authFlow.authState);
```

Comment:
Can we verify that this `authState` is what we expect it to be?

Reply:
Done.